cpp/demo/biharmonic/main.cpp

```diff
@@ -1,124 +1,160 @@
+// $$
 //
-// where :math:`\nabla^{4} \equiv \nabla^{2} \nabla^{2}` is the
-// biharmonic operator and :math:`f` is a prescribed source term. To
-// formulate a complete boundary value problem, the biharmonic equation
+// where $\nabla^{4} \equiv \nabla^{2} \nabla^{2}$ is the biharmonic operator
+// and $f$ is a prescribed source term.
+// To formulate a complete boundary value problem, the biharmonic equation
 // must be complemented by suitable boundary conditions.
 //
+// ### Weak formulation
+//
 // Multiplying the biharmonic equation by a test function and integrating
-// by parts twice leads to a problem second-order derivatives, which
-// would requires :math:`H^{2}` conforming (roughly :math:`C^{1}`
-// continuous) basis functions.  To solve the biharmonic equation using
-// Lagrange finite element basis functions, the biharmonic equation can
-// be split into two second-order equations (see the Mixed Poisson demo
-// for a mixed method for the Poisson equation), or a variational
-// formulation can be constructed that imposes weak continuity of normal
-// derivatives between finite element cells.  The demo uses a
-// discontinuous Galerkin approach to impose continuity of the normal
-// derivative weakly.
-//
-// Consider a triangulation :math:`\mathcal{T}` of the domain
-// :math:`\Omega`, where the set of interior facets is denoted by
-// :math:`\mathcal{E}_h^{\rm int}`.  Functions evaluated on opposite
-// sides of a facet are indicated by the subscripts ':math:`+`' and
-// ':math:`-`'.  Using the standard continuous Lagrange finite element
-// space
-//
-// .. math::
-//     V = \left\{v \in H^{1}_{0}(\Omega)\,:\, v \in P_{k}(K) \ \forall \ K \in
-//     \mathcal{T} \right\}
+// by parts twice leads to a problem of second-order derivatives, which would
+// require $H^{2}$ conforming (roughly $C^{1}$ continuous) basis functions.
+// To solve the biharmonic equation using Lagrange finite element basis
+// functions, the biharmonic equation can be split into two second-order
+// equations (see the Mixed Poisson demo for a mixed method for the Poisson
+// equation), or a variational formulation can be constructed that imposes
+// weak continuity of normal derivatives between finite element cells.
+// This demo uses a discontinuous Galerkin approach to impose continuity
+// of the normal derivative weakly.
+//
+// Consider a triangulation $\mathcal{T}$ of the domain $\Omega$, where
+// the set of interior facets is denoted by $\mathcal{E}_h^{\rm int}$.
+// Functions evaluated on opposite sides of a facet are indicated by the
+// subscripts $+$ and $-$.
+// Using the standard continuous Lagrange finite element space
+//
+// $$
+// V = \left\{v \in H^{1}_{0}(\Omega)\,:\, v \in P_{k}(K) \
+// \forall \ K \in \mathcal{T} \right\}
+// $$
 //
 // and considering the boundary conditions
 //
-// .. math::
-//    u            &= 0 \quad {\rm on} \ \partial\Omega \\
-//    \nabla^{2} u &= 0 \quad {\rm on} \ \partial\Omega
+// $$
```

Comment:
Would be better to use the amsmath environments rather than $$

https://myst-parser.readthedocs.io/en/v0.14.0/using/syntax-optional.html#syntax-amsmath

Reply:
One could use both $$ and amsmath simultaneously.